internal/api/v20240610preview/hcpopenshiftclusters_methods.go

```diff
@@ -319,45 +318,53 @@ func (c *HcpOpenShiftClusterResource) Normalize(out *api.HCPOpenShiftCluster) {
 func validateStaticComplex(normalized *api.HCPOpenShiftCluster) []arm.CloudErrorBody {
 	var errorDetails []arm.CloudErrorBody
-	//Generate a slice of Resource IDs of ControlplaneOperators MI only , disregard the DataPlaneOperators.
-	var cpmiResourceIDs []string
+
 	if normalized.Identity.UserAssignedIdentities != nil {
+		//Initiate the map that will have the number occurence of resource IDs in the Identity and Operator Authn fields .
+		identityOccurrences := make(map[string]int)
+		operatorAuthnOccurrences := make(map[string]int)
+		//Generate a Map of Resource IDs of ControlplaneOperators MI and Service MI if exists , disregard the DataPlaneOperators.
 		for _, operatorResourceID := range normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ControlPlaneOperators {
-			cpmiResourceIDs = append(cpmiResourceIDs, operatorResourceID)
+			operatorAuthnOccurrences[operatorResourceID]++
 		}
 		smiResourceID := normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ServiceManagedIdentity
-		miCount := len(normalized.Identity.UserAssignedIdentities)
-		for resourceID, _ := range normalized.Identity.UserAssignedIdentities {
-
-			if slices.Contains(cpmiResourceIDs, resourceID) || resourceID == smiResourceID {
-				//Duplicated Identity Validation
-				if slices.Contains(cpmiResourceIDs, resourceID) && resourceID == smiResourceID {
-					errorDetails = append(errorDetails, arm.CloudErrorBody{
-						Message: fmt.Sprintf(
-							"%s can not be used in both controlPlaneOperators and serviceManagedIdentity",
-							resourceID),
-						Target: "properties.spec.platform.operatorsAuthentication.userAssignedIdentities",
-					})
-				}
-			} else {
+
+		if smiResourceID != "" {
+			operatorAuthnOccurrences[smiResourceID]++
+		}
+
+		for resourceID := range normalized.Identity.UserAssignedIdentities {
+			identityOccurrences[resourceID]++
+		}
+		for resourceID := range operatorAuthnOccurrences {
+			_, ok := identityOccurrences[resourceID]
+			if !ok {
 				errorDetails = append(errorDetails, arm.CloudErrorBody{
 					Message: fmt.Sprintf(
-						"%s must be used in either controlPlaneOperators or serviceManagedIdentity",
+						"Identity %s is referenced in Operator Authentication but is not defined in UserAssignedIdentities",
 						resourceID),
-					Target: "properties.spec.platform.operatorsAuthentication.userAssignedIdentities"})
+					Target: "properties.spec.platform.operatorsAuthentication.userAssignedIdentities",
+				})
+			} else if identityOccurrences[resourceID] != operatorAuthnOccurrences[resourceID] {
+				errorDetails = append(errorDetails, arm.CloudErrorBody{
+					Message: fmt.Sprintf(
+						"Identity %s appears %d times in Operator Authentication but %d times in UserAssignedIdentities", resourceID, operatorAuthnOccurrences[resourceID], identityOccurrences[resourceID]),
+					Target: "properties.spec.platform.operatorsAuthentication.userAssignedIdentities",
+				})
+
 			}
```

Comment:
I think you could maybe drop this check.  Instead, iterate over `operatorAuthOccurrences` and look for any 2 values (it's either going to be 1 or 2).